book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -524,49 +524,62 @@ The state service uses the following entry points:
 
 ## Committing non-finalized blocks
 
-Given the above structures for manipulating the non-finalized state new
-`non-finalized` blocks are commited as follows:
+New `non-finalized` blocks are commited as follows:
 
-### `fn queue_and_commit_non_finalized_blocks(&mut self, new: QueuedBlock)`
+### `pub(super) fn queue_and_commit_non_finalized_blocks(&mut self, new: Arc<Block>) -> tokio::sync::broadcast::Receiver<block::Hash>`
 
-1. If the block itself exists in the finalized chain, it has already been
-   successfully verified:
-  - broadcast `Ok(block.hash())` via `block.rsp_tx`, and return
+1. If a duplicate block exists in the queue:
+     - Find the `QueuedBlock` for that existing duplicate block
+     - Create an extra receiver for the existing block, using `block.rsp_tx.subscribe`,
+     - Drop the newly received duplicate block
+     - Return the extra receiver, so it can be used in the response future for the duplicate block request
```

Comment:
We can split this out into a new function `check_for_queued_duplicate_block`, which:
- takes `Arc<Block>`
- if a duplicate is found, returns the extra receiver

(We don't have to make this change in the spec, it's enough to just make it in the code.)